oak_functions/client/rust/src/rekor.rs

```diff
@@ -159,10 +206,85 @@ pub fn verify_rekor_signature(
     pem_encoded_public_key_bytes: &[u8],
 ) -> anyhow::Result<()> {
     let signature_bundle = rekor_signature_bundle(log_entry_bytes)?;
+
+    verify_signature(
+        &signature_bundle.base64_signature,
+        &signature_bundle.canonicalized,
+        pem_encoded_public_key_bytes,
+    )
+    .context("failed to verify signedEntryTimestamp of the Rekor LogEntry")
+}
+
+/// Verifies the signature in the `body` over the `endorsement_bytes`, using the public key in
+/// `pem_encoded_public_key_bytes`.
+///
+/// Returns `Ok(())` if the verification succeeds, otherwise returns `Err()`.
+pub fn verify_rekor_body(
+    body: &Body,
+    endorsement_bytes: &[u8],
+    pem_encoded_public_key_bytes: &[u8],
+) -> anyhow::Result<()> {
+    if body.spec.signature.format != "x509" {
+        anyhow::bail!(
+            "unsupported signature format: {}; only x509 is supported",
+            body.spec.signature.format
+        )
+    }
+
+    // For now, we only support `sha256` as the hashing algorithm.
+    if body.spec.data.hash.algorithm != "sha256" {
+        anyhow::bail!(
+            "unsupported hash algorithm: {}; only sha256 is supported",
+            body.spec.data.hash.algorithm
+        )
+    }
+
+    // Check that hash of the endorsement statement matches the hash of the data in the Body.
+    let endorsement_hash = get_sha256(endorsement_bytes);
+    let endorsement_hash_hex = hex::encode(endorsement_hash);
+    if endorsement_hash_hex != body.spec.data.hash.value {
+        anyhow::bail!(
+            "the hash of the endorsement file ({:?}) does not match the hash of the data in the body of the rekor entry ({:?})",
+            endorsement_hash_hex,
+            body.spec.data.hash.value
+        )
+    }
+
+    // Check that the public key in the body matches the given public key. This in fact checks the
```

Comment:
clarification: the "given public key". I see it is the argument of `verify_rekor_body`, but where did we actually get the key from in the first place?